### DISCUSSION TITLE
linkQQ发送绑定问题

### DISCUSSION THREAD

**Discussion by TilyTao** [DreamVoid/MiraiMC#202] [问答 / Q&A]
从qq发到游戏内：如添加绑定 abc
机器人没有任何反应,不会出现绑定码

在游戏内使用link bind 123456，出现绑定码后，qq内输入确认绑定 机器人没有任何回应，会报错
**Could not pass event MiraiGroupMessageEvent to Chat2QQ v1.6
java.lang.NoSuchMethodError: 'java.util.UUID me.dreamvoid.miraimc.api.MiraiMC.getBind(long)'**

MiraiMC用的是。1.7

同时，用/miraimc bind add <玩家名> <QQ号> ，也会报错
/miraimc bind add 游戏名 QQ号
[10:41:36 WARN]: [MiraiMC] Plugin MiraiMC v1.7-rc1 generated an exception while executing task 29992
java.lang.IllegalArgumentException: Invalid UUID string: xxxxx
        at java.util.UUID.fromString1(UUID.java:280) ~[?:?]
        at java.util.UUID.fromString(UUID.java:258) ~[?:?]

在QQchat里配置文件
  # 是否必须使用命令才能向QQ群发送聊天消息
  require-command-to-chat: false
设置成false，无法同步服务器内的聊天到群里，只能使用/qchat 发送才能到群里

linkqq 修改配置文件后重新加载无效，只能重启服务端

**Comment by TilyTao** [comment]
服务端为：paper1.18.2
chatqq：1.6
miraimc:[1.7-rc1]
linkqq：1.1

**Comment by Muska-Ami** [comment]
xd这是MiraiMC储存库，还有不能更新下rc-2吗

**Reply by Muska-Ami** [reply]
还有MiraiMC1.7-pre2改了事件
为兼容依赖旧版本MiraiMC的插件，请将 config.yml 中的 legacy-event-support 项设为 true

**Reply by TilyTao** [reply]
> 

legacy改成true可以用linkqq了，chatQQ依旧出现'java.util.UUID me.dreamvoid.miraimc.api.MiraiMC.getBind(long)

**Reply by Muska-Ami** [reply]
能发一下/ver出来的东西吗

**Reply by TilyTao** [reply]
> 能发一下/ver出来的东西吗
![image](https://user-images.githubusercontent.com/19508889/178869455-519c130d-498e-486e-897a-47d17cd78bb3.png)
这玩意？

**Reply by Muska-Ami** [reply]
这我也不知道咋办，我用MiraiMC-rc2+Chat2QQ，1.18.2Purpur端无问题，另外我写的
https://github.com/XiaMoHuaHuo-CN/CatSero
里也提供了一个聊天转发，不行你再试试

**Reply by TilyTao** [reply]
> 这我也不知道咋办，我用MiraiMC-rc2+Chat2QQ，1.18.2Purpur端无问题，另外我写的 https://github.com/XiaMoHuaHuo-CN/CatSero 里也提供了一个聊天转发，不行你再试试

哈哈，好的，我这就试试，这几天被逼的，我都开始自己写了，一个小功能都出来了.............

**Reply by TilyTao** [reply]
> > 这我也不知道咋办，我用MiraiMC-rc2+Chat2QQ，1.18.2Purpur端无问题，另外我写的 https://github.com/XiaMoHuaHuo-CN/CatSero 里也提供了一个聊天转发，不行你再试试
> 
> 哈哈，好的，我这就试试，这几天被逼的，我都开始自己写了，一个小功能都出来了.............

 发送GET请求出现异常:
java.net.ConnectException: Connection timed out: connect[java.base/sun.nio.ch.Net.connect0(Native Method), java.base/sun.nio.ch.Net.connect(Net.java:579), java.base/sun.nio.ch.Net.connect(Net.java:568), java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:588), java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327), java.base/java.net.Socket.connect(Socket.java:639), java.base/sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:304), java.base/sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:174), java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:183), java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:498), java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:603), java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:266), java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:380), java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:189), java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1245), java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1131), java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:175), java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:142), CatSero-1.1.2-ifx1.jar//moe.xmcn.catsero.utils.HttpUtils.sendGet(HttpUtils.java:36), CatSero-1.1.2-ifx1.jar//moe.xmcn.catsero.Updater.startUpdateCheck(Updater.java:32), CatSero-1.1.2-ifx1.jar//moe.xmcn.catsero.Main$1.run(Main.java:62), org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftTask.run(CraftTask.java:101), org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57), com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635), java.base/java.lang.Thread.run(Thread.java:833)]
[09:57:37 WARN]: [CatSero] Plugin CatSero v1.1.2-ifx1 generated an exception while executing task 18
java.lang.NullPointerException: Cannot invoke "String.contains(java.lang.CharSequence)" because "nowversion" is null
        at moe.xmcn.catsero.Updater.startUpdateCheck(Updater.java:38) ~[CatSero-1.1.2-ifx1.jar:?]
        at moe.xmcn.catsero.Main$1.run(Main.java:62) ~[CatSero-1.1.2-ifx1.jar:?]
        at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.18.2.jar:git-Paper-387]
        at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.18.2.jar:git-Paper-387]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.18.2.jar:git-Paper-387]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

**Reply by Muska-Ami** [reply]
你网不好没法更新别管这玩意（）

**Reply by TilyTao** [reply]
> 你网不好没法更新别管这玩意（）

启用转发好像并不能转发，还有加入游戏和退出游戏事件
# 转发MC-QQ聊天
forward-chat:
  # 启用此功能
  enabled: true
  # 格式
  format:
    to-mc: "[%time%][QQ][%groupname%(%groupcode%)]%sendername%(%sendercode%):\n%message%"
    to-qq: "[MC]%player%:\n%message%"
  prefix:
    # 启用此功能
    enabled: false
    # 格式(只支持单个，字符，请勿输入多个)
    format:
      to-mc: "#"
      to-qq: "#"
  # 发到QQ自动清理消息中的颜色代码
  clean-colorcode: true
  # 使用MiraiMC内置绑定
  use-bind: false
没有转发任何信息

**Reply by Muska-Ami** [reply]
到时候我再debug一下吧，这个可能我改底层的时候搞炸了

**Reply by Muska-Ami** [reply]
另外要在config.yml配置bot和group

**Reply by TilyTao** [reply]
....前面的版本能不能用

**Reply by Muska-Ami** [reply]
1.1你试试？，或者1.1.1

**Reply by Muska-Ami** [reply]
1.1你试试？，或者1.1.1
顺便给我开个Issue
https://github.com/XiaMoHuaHuo-CN/CatSero/issues

**Reply by TilyTao** [reply]
换成1.1和最新的一样，都不转发消息

**Reply by Muska-Ami** [reply]
是这个包

**Reply by TilyTao** [reply]
> 是这个包

哎，怎么都不转发消息，很蛋疼

**Reply by Muska-Ami** [reply]
你监听到消息然后直接全局发送MC消息不就好了

**Reply by TilyTao** [reply]
是的，正在做这件事，我先自己写个简单的用着吧
